tools/acorn-optimizer.js

```diff
@@ -379,10 +383,18 @@ function runJSDCE(ast, aggressive) {
           c(param.right);
           param = param.left;
         }
-        assert(param.type === 'Identifier', param.type);
-        const name = param.name;
-        ensureData(scope, name).def = 1;
-        scope[name].param = 1;
+        if (param.type === 'ArrayPattern') {
+          for (var elem of param.elements) {
+            markLocal(elem.name);
```

Comment:
Can you change `(param) => ` above to `function traverse(param) {`? Then you could change these things to `return traverse(elem);` which would both handle what you have now, as well as would correctly handle nested patterns (which it doesn't yet).

Reply:
Done.  Although I don't know about the `return`  in `return traverse(elem);` surely that wouldn't work because need to loop over each elem in the list right?

Reply:
Ah yeah, sorry. It would work if you used `forEach` but you're right, not with a regular loop.